src/main/java/ui/windows/StatsController.java

```diff
@@ -23,32 +25,35 @@ public void setDisplay(StatsDisplay display) {
 
     /**
      * Requests statistics for a given user and updates display.
+     * @param player the PlayerInformation of the user
      * @param username the name of the user
      * @return false if no display is supplied, true otherwise
      */
-    public boolean tryRequestUser(String username) {
+    public boolean tryRequestUser(PlayerInformation player, String username) {
         if (display == null) {
             return false;
         }
 
-        StatsDisplayData data = retrieveData(username);
+        StatsDisplayData data = retrieveData(player, username);
         display.updateView(data);
         return true;
     }
 
     /**
      * Retrieves statistics for a given user.
+     * @param player the PlayerInformation of the user
      * @param username the name of the user
      * @return the retrieved data as a StatsDisplayData
      */
-    private StatsDisplayData retrieveData(String username) {
-        // TODO: send usecase 
-
-        StatsDisplayData data = new StatsDisplayData();
-        data.setName(username);
-        data.setGamesPlayed(0); // TODO: replace with retrieved data
-        data.setGamesWon(0); // ditto
-        data.setLongestWinStreak(0); // ditto ditto
+    private StatsDisplayData retrieveData(PlayerInformation player, String username) {
+        StatsDisplayData data;
+
+        if (player != null) {
+            data = new StatsDisplayData(player.getName(), player.getWins(), player.getLosses());
```

Comment:
id suggest calling a use case that interacts with the database here, that retrieves the appropriate information from the database

you can store a reference to the use case as an attribute in this class

Reply:
PlayerInformation is in use_cases so I assumed this was fine? The only other file in use_cases is the DataAccess interface which will require me to use PlayerInformation anyways and the getName(), getWins(), and getLosses() methods to access the information.

Reply:
importing PlayerInformation is fine, its just the method you retrieve the data

youre getting the data from a PlayerInformation thats already passed in, but the controller needs to request the data from a database